### ISSUE TITLE
Library maturity / coverage

### ISSUE THREAD

**Issue by sumanyu** [haifengl/smile#111]
This is really a question.

I'm quite familiar with scikit-learn in python. It has excellent documentation and coverage of most popular ML algorithms.

I am looking for a JVM / scala equivalent and this seems to come close, although still being actively developed. I was searching for Scala documentation / demos of this but they are incomplete / in java. 

Are you moving towards scala bindings as primary? When are you planning on adding scikit-learn like documentation for the project?

**Comment by haifengl** [comment]
You probably should ask this question on [gitter](https://gitter.im/haifengl/smile) to people using smile as my answer is surely biased :)

Smile is actually very mature and all algorithms are covered by many test cases. The Javadoc and Scaladoc documents are also very intensive. I am not sure what do you mean scikit-learn like documentation. But you can find more tutorial on http://haifengl.github.io/smile/

BTW, we cover many more algorithms than scikit-learn and we are planning more (on deep learning). I don't understand why you feel a project is not mature if it is being actively developed. In my opinion, a project without development is a dead project.

**Comment by sumanyu** [comment]
Cool! Will do.

I clicked on User Guide and basically found none of the links were actually active, so I was judging the book by its cover. I haven't dug too deep into the scala docs yet.

Let me rephrase. There is a difference between bleeding edge and leading edge. I couldn't quite tell whether the core is stable and new features are added or the core apis are rapidly changing. Hence, I asked! :)

**Comment by haifengl** [comment]
Did you view the user guide on iPad? The menu doesn't go away after clicking on iPad although the content is updated. Sorry for the issue. Try on a computer.

**Comment by haifengl** [comment]
Most algorithms have been implemented for several years. They are really mature.

Btw, all major algorithms are in Java. Try java doc first.
